### PR TITLE
Fixed #26196 -- Made sure __in lookups use to_field as default.

### PR DESCRIPTION
**Pull request by charettes**
Thanks to Trac alias hellpain for the report.

### CODE REVIEW

File: django/db/models/fields/related_lookups.py [hunk 1]
```diff
@@ -79,6 +79,16 @@ def as_sql(self, compiler, connection):
                         [source.name for source in self.lhs.sources], self.rhs),
                     AND)
             return root_constraint.as_sql(compiler, connection)
+        elif not self.rhs_is_direct_value():
+            from django.db.models.sql.where import WhereNode, SubqueryConstraint, AND, OR
+            root_constraint = WhereNode(connector=OR)
+            root_constraint.add(
+                SubqueryConstraint(
+                    self.lhs.alias, [self.lhs.target.column],
+                    [f.get_attname_column()[1] for f in self.lhs.output_field.get_path_info()[0].target_fields],
```
Comment (charettes) [inline django/db/models/fields/related_lookups.py]: @akaariai I'm really not sure from where I should retrieve the `SubqueryConstraint.targets`. I see that `self.lhs.sources` is available when `isinstance(self.lhs, MultiColSource)` but in this case `self.lhs.sources` is an empty list.

**Comment by hellpain** [comment]
As I see not all django test are passing, so this PR solves one problem and creates another)

File: django/db/models/fields/related_lookups.py (hunk 1 above)
Comment (akaariai) [inline django/db/models/fields/related_lookups.py]: I'm going to try to work on Django tomorrow. I'll try to work on this.

**Comment by timgraham** [comment]
Updated by Anssi in https://github.com/django/django/pull/6117.

**Pull request closed without merging**